subprojects/model-core/src/main/java/org/gradle/api/internal/provider/DefaultValueSourceProviderFactory.java

```diff
@@ -172,6 +173,11 @@ public boolean isValueProducedByTask() {
             return value == null;
         }
 
+        @Override
+        public boolean immutable() {
+            return true;
+        }
```

Comment:
Why not use `isImmutable()` here?

Reply:
It's implementing `Managed#immutable`, you got a point. I'll see if it's ok to rename the whole hierarchy.